src/planscape/forsys/get_forsys_inputs.py

```diff
@@ -94,35 +149,186 @@ def __init__(
-                len(priorities), len(base_condition_ids_to_names.keys())))
-        return base_condition_ids_to_names
-
-    def _get_conditions(self, condition_ids: list[int]) -> list[Condition]:
-        conditions = list(Condition.objects.filter(
-            condition_dataset_id__in=condition_ids).filter(
-            is_raw=False).all())
-        if len(condition_ids) != len(conditions):
-            raise Exception(
-                "of %d priorities, only %d had conditions" %
-                (len(condition_ids),
-                 len(conditions)))
-        return conditions
-
-    def _get_initialized_forsys_input(self, headers: ForsysInputHeaders,
-                                      priorities: list[str]) -> dict[str, list]:
-        forsys_input = {}
-        forsys_input[headers.FORSYS_PROJECT_ID_HEADER] = []
-        forsys_input[headers.FORSYS_STAND_ID_HEADER] = []
-        forsys_input[headers.FORSYS_AREA_HEADER] = []
-        forsys_input[headers.FORSYS_COST_HEADER] = []
-        for p in priorities:
-            forsys_input[headers.get_priority_header(p)] = []
-        return forsys_input
+
+class ForsysGenerationInput():
+    # Treatment cost per kilometer-squared (in USD)
+    # TODO: make this variable based on a user input and/or a treatment cost
+    # raster.
+    TREATMENT_COST_PER_KM_SQUARED = 5000 * 1000 * 1000
+
+    # A dictionary representing a forsys input dataframe.
+    # In the dataframe, headers correspond to ForsysInputHeaders headers. Each
+    # row represents a unique stand.
+    # Dictionary keys are dataframe headers. Dictionary values are lists
+    # corresponding to columns below each dataframe header.
+    forsys_input: dict[str, list]
+
+    # ----- Intermediate data -----
+    # Maps condition names to retrieved ConditionPixelValues instances.
+    _conditions_to_raster_values: dict[str, ConditionPixelValues]
+    # The origin coordinate used when merging ConditionPixelValues instances, which may have different top-left coordinates, across all conditions.
+    _topleft_coords: tuple[float, float]
+    # Raw condition values merged via embedded dictionaries keyed by x pixel
+    # index, y pixel index, and condition name.
+    _pixel_dist_x_to_y_to_condition_to_values = dict[int,
+                                                     dict[int,
+                                                          dict[str, float]]]
+
+    def __init__(
+            self, params: ForsysGenerationRequestParams,
+            headers: ForsysInputHeaders) -> None:
+        region = params.region
+        priorities = params.priorities
+        planning_area = params.planning_area
+
+        self.forsys_input = _get_initialized_forsys_input_with_common_headers(
+            headers, priorities)
+        self.forsys_input[headers.FORSYS_GEO_WKT_HEADER] = []
+
+        geo = get_raster_geo(planning_area)
+
+        self._fetch_condition_raster_values(geo, priorities, region)
```

Comment:
Yes. Done. I've made all three functions explicitly return intermediate objects rather than implicitly populate them.

I was hesitant to do this originally because the idea of copying dictionaries every time a new row is populated is gross - but then I realized Dictionaries are passed by reference in python.

Reply:
makes sense :)